### ISSUE TITLE
Bazaar plugin Database migration failure

### ISSUE THREAD

**Issue by tviaud** [backstage/backstage#8276]
Hi all 👋 
I have a little issue with a DB migration for the bazaar-backend plugin.

## Expected Behavior
The bazaar plugin should migrate to the new Database schema successfully with existing data.

## Current Behavior
The migration is failing with the following message:
```
migration failed with error: alter table "metadata" add column "size" text not null default 'medium', add column "start_date" text, add column "end_date" text, add column "responsible" text not null - column "responsible" of relation "metadata" contains null values
```

The table for the bazaar-backend plugin contains one item with the following columns: `entity_ref,name,community,announcement,status,updated_at`

## Steps to Reproduce

1. Add bazaar projects items with version 0.1.1 of plugin-bazaar-backend.
1. Migrate from plugin-bazaar-backend:0.1.1 to 0.1.2

## Context

Rolling out a new version of our Backstage instance.
This Backstage instance contained the updated version of the bazaar-backend plugin version `0.1.2` migrating from `0.1.1`.
A database migration is trying to take place but fails.
Our database contains one item that has been created in version 0.1.1 and fails to migrate the table and add the new columns introduced in `0.1.2`.
## Your Environment
Backstage instance with plugin-bazaar-backend `0.1.2`.

Postgres database version `13.3`.

- NodeJS Version (v14): v14
- Operating System and Version (e.g. Ubuntu 14.04): docker image: node:14-buster-slim

**Comment by freben** [comment]
I tried making a fix for this in #8337 - @tviaud If you are up for it, you could locate the fixed file from that PR inside your node_modules folder, add that same line, and let us know if that seems to fix the issue. This is of course just temporary and informative for us, rather than a proper fix :) The actual fix will be released on Thursday next week, unfortunately.

**Comment by tviaud** [comment]
Hi @freben , I'm sorry but I already migrated to the new table by removing the affected row.
I won't be able to test it :(
Thank you for the quick patch!
